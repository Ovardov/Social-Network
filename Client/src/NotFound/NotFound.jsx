import React from 'react';
import { Link } from 'react-router-dom';
import styles from './not-found.module.scss';

function NotFound() {
    return (
        <section className={styles.container}>

            <div className={styles.info}>
                <h2>404 - Page Not Found</h2>
                <Link className="button" to="/">Go to Homepage</Link>
            </div>

            <img src="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576342536/o7hfrt963jvu4oqzdbov.png" alt="404" />
        </section>
    )
}

export default NotFound;