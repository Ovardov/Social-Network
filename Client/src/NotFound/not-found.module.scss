.container {
    display: flex;
    justify-content: space-around;
    align-items: center;
    background: -webkit-linear-gradient(40deg, #FFFFFF 50%, var(--color-blue-main));
    height: 100vh;
    width: 100%;
    color: var(--color-blue-main);
    text-align: center;
    font-size: 1.5em;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 9999;
}