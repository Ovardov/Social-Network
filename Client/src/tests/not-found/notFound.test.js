import React from 'react'
import renderer from 'react-test-renderer';
import NotFound from '../../components/NotFound/NotFound';


jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('NotFound component', () => {
    test('should render correctly', () => {
        const component = renderer.create(
            <NotFound />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
})
