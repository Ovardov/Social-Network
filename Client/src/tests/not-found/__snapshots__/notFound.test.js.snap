// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render correctly 1`] = `
<section
  className="container"
>
  <div
    className="info"
  >
    <h2>
      404 - Page Not Found
    </h2>
    <Link
      className="button"
      to="/"
    >
      Go to Homepage
    </Link>
  </div>
  <img
    alt="404"
    src="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576342536/o7hfrt963jvu4oqzdbov.png"
  />
</section>
`;
